Source/DafnyLanguageServer/Language/DiagnosticErrorReporter.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.Boogie;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.Dafny.LanguageServer.Language
+{
+  public class DiagnosticErrorReporter : ErrorReporter
+  {
+    private const MessageSource verifierMessageSource = MessageSource.Other;
+    private readonly Dictionary<string, List<Diagnostic>> diagnostics = new();
+    private readonly Dictionary<DiagnosticSeverity, int> counts = new();
+
+    public IReadOnlyDictionary<string, List<Diagnostic>> Diagnostics => diagnostics;
+
+    public void ReportBoogieError(ErrorInformation error)
+    {
+      var tok = error.Tok;
+      var relatedInformation = new List<DiagnosticRelatedInformation>() { };
+      foreach (var auxErrorInfo in error.Aux)
+      {
+        if (auxErrorInfo.Category == "Related location") {
+          relatedInformation.Add(new DiagnosticRelatedInformation()
+          {
+            Message = auxErrorInfo.Msg,
+            Location = new Location()
+            {
+              Range = auxErrorInfo.Tok.GetLspRange(),
+              
+              // https://github.com/dafny-lang/dafny/blob/06b498ee73c74660c61042bb752207df13930376/Source/DafnyLanguageServer/Language/DafnyLangParser.cs#L59
```

Comment:
Can you make this a "see also" of some kind that doesn't require a web browser to follow/understand?